app/models/application_search_filter.rb

```diff
@@ -22,96 +21,116 @@ def assigned_status_options
     status_options + assigned_to_user_options
   end
 
-  #
   # Options for the application status filter
-  #
   # Includes 'Open', 'Completed', 'Sent back to provider' or 'All applications'
   # Values can be "open", "completed", "sent_back", "all"
-  #
   def application_status_options
     Types::REVIEW_STATUS_GROUPS.keys.map do |status|
       [I18n.t(status, scope: 'values.review_status'), status]
     end
   end
 
-  #
-  # Hash of DatastoreApi search constraints based upon the filter
-  #
-  # NOTE: The DatastoreApi does not understand assigned_status. To filter
-  # DatastoreApi searches by assigned status, we translate the assigned_status into
-  # "application_id_in" and "application_id_not_in" DatastoreApi constraints.
-  #
+  # Options for the application age_in_business_days filter
+  # Includes '0 days', '1 day', '2 days', '3 days'
+  def age_in_business_days_options
+    Array.new(4) do |day|
+      [I18n.t('values.days_passed', count: day), day]
+    end
+  end
+
+  # Hash of DatastoreApi search constraints based on the active filters
   def datastore_params
-    {
-      applicant_date_of_birth: applicant_date_of_birth,
-      application_id_in: application_id_in,
-      application_id_not_in: application_id_not_in,
-      review_status: review_status,
-      submitted_after: submitted_after&.in_time_zone('London'),
-      submitted_before: submitted_before&.in_time_zone('London'),
-      search_text: search_text
-    }
+    active_datastore_filters.each_with_object({}) do |filter, params|
+      params.merge!(send("#{filter}_datastore_param"))
+    end
+  end
+
+  # If there are no applications that satisfy the active review filters,
+  # there is no need to perform a datastore query as the results will be empty.
+  def datastore_results_will_be_empty?
+    !active_review_filters.empty? && active_review_filters_application_ids.empty?
   end
 
   private
 
-  def assigned_to_user_options
-    # builds an array of [User#name, User#id] sorted by first name, last name
-    ids_of_users_with_active_assignments_and_or_reviews.map { |id| [User.name_for(id), id] }.sort
+  def active_filters
+    attributes.compact_blank.keys
   end
 
-  def ids_of_users_with_active_assignments_and_or_reviews
-    (
-      CurrentAssignment.distinct.pluck(:user_id) + Review.distinct.pluck(:reviewer_id)
-    ).compact.uniq
-  end
-
-  # Returns the value of the DatastoreApi Search "application_id_in" constraint
-  # according to the #assigned_status. Assigned status can be either nil, 'unassigned',
-  # 'assigned' or a user_id.
-  #
-  # When assigned_status is a user_id, it returns the application_ids of those
-  # applications that are either currently assigned to or were reviewed by
-  # the given user.
-  def application_id_in
-    case assigned_status
-    when nil, 'unassigned'
-      []
-    when 'assigned'
-      all_assigned_application_ids
-    else
-      application_ids_for_user(assigned_status)
-    end
+  def active_review_filters
```

Comment:
Thinking this might be more clear as `unique_active_review_filters` - if they can exist in both active filters and REVIEW_FILTERS they I understand the need for `&` if they cant then better to just concatenate the arrays.